Docs: sync the (partial) release notes for the next LTS

clebergnu opened the pull request:
This puts the relevant changes from version 56.0 to 59.0 into the
(partial, ongoing) list of changes from the previous LTS version.

Signed-off-by: Cleber Rosa <crosa@redhat.com>

clebergnu commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add partial LTS related changes to docs from 56.0 to 59.0](https://trello.com/c/OS48IVTl/1199-add-partial-lts-related-changes-to-docs-from-560-to-590)

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
   might still modify this to better follow their conventions (eg.
   for RPM the original locations are used)
```
Hmm, it's not part of this PR, but it'd be nice to include the hierarchy of those files (to safe time to people that are reading it) as even `/etc` is respected.

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  at the end of job execution.  Please refer to
+  :ref:`results-upload-plugin` for more information.
+
+* The ``default_parameters`` mechanism for setting default parameters
```
I think it'd help to explicitly mention `The "avocado.Test.default_parameters" variable which used to be used to set default params...` (at lease I'd like to see the `avocado.Test.default_parameters` there as it's not always obvious what feature and where it used to be located)

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+
+* The ``default_parameters`` mechanism for setting default parameters
+  on tests has been removed.  This was introduced quite early in the
+  Avocado development, and allowed users to set a dictionary at the
```
I don't think it's necessary to include explanation. It was simply removed (anyway I'm not strictly against, I just don't think it adds anything useful to users reading it).

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  Avocado development, and allowed users to set a dictionary at the
+  class level with keys/values that would serve as default parameter
+  values.  The recommended approach now, is to just provide default
+  values when calling ``self.parameters.get`` within a test method,
```
`self.params.get`

clebergnu replied:
Yep, you're right.

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  is ``self.params.get("key")`` to achieve the same thing.
+
+* The :mod:`avocado.utils.vmimage` library now contains support for
+  Avocado's own JeOS image.  A nice addition given the fact that
```
I think short description like `JeOS [Just Enough Operation System - Stripped-out Fedora with minimal deps]`. It's not long and can promote the concept. People don't know they need it :smile:

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  Avocado's own JeOS image.  A nice addition given the fact that
+  it's the default image used in Avocado-VT and the latest version
+  is available in the following architectures: x86_64, aarch64, ppc64,
+  ppc64le and s390x.
```
... and is regularly updated (max skip is usually 1-2 releases)

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  are now considered package data files of the ``avocado`` package.
+  The end result is that installation from source works fine outside
+  virtual environments (in addition to installations *inside* virtual
+  environments).
```
This was partially described on lines 83 - 89. I think you should just update that one and not adding new bullets.

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  tests based on the `GLib test framework
+  <https://developer.gnome.org/glib/stable/glib-Testing.html>`_.  This
+  plugin allows individual tests inside a single binary to be listed
+  and executed.
```
I see many new plugins. What would you say about adding new bullet-list with new features and group plugins/optional_plugins in there?

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  names.  This allows users to easily differentiate among execution of
+  the same test, but executed different configurations.  For more
+  information, look for "special keys" in the :ref:`YAML Loader plugin
+  documentation <yaml_loader>`.
```
And I think it'd help putting plugin-updates in separate bullet list, perhaps prefixed with plugin they belong to (it took me a while to realize what are you talking about)

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  when users do not manually dump a variants file to a specific
+  location, Avocado will automatically save a suitable file at
+  ``jobdata/variants.json`` as part of a Job results directory
+  structure.
```
Do you think we should mention the limitation to JSON compatible variables only? I'm not sure it belongs to changelog, but you are quite verbose so a note about that might help.

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+  :attr:`avocado.utils.process.CmdResult.stderr_text`, which are
+  convenience properties that will attempt to decode the ``stdout`` or
+  ``stderr`` data into a string-like type using the encoding set, and
+  if none is set, falling back to the system default encoding.  This
```
python-default encoding (for now)

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+* The TAP result format plugin received improvements, including
+  support for reporting Avocado tests with CANCEL status as SKIP
+  (which is the closest status available in the TAP specification),
+  and providing more visible warning information in case Avocado tests
```
how about mentioning the `comment` in the sentence? I mean that is how warnings are passed.

clebergnu replied:
Done.

ldoktor commented on docs/source/release_notes/lts/next.rst:
```diff
@@ -88,6 +88,118 @@ introduced by the next LTS version are:
+
+* Redundant (and deprecated) fields in the test sections of the JSON
+  result output were removed.  Now, instead of ``url``, ``test`` and
+  ``id`` carrying the same information, only ``id`` remains.
```
Would you please add following:

* d2458c6dc5121384bd78c0768b8f040ed037f328 - new arguments
* bc59f8506fc4700a111d66ac3a6ab1024f4e8c25 - removed 36 compatibility
* 4eed345e4bd59993e03c3e2d21098ee9f0ea5a57 - new functions in multipath
* 9b9a335c3fe548348deffe15fe528ac4b0e2a3a9 - subtest filter for dirs
* 27661323f98e4fb8172f39b9359ae567ce11a003 - subtest filter for simpletests
* b38129958c0c21c0c725b8cea021dc06398d6b71 - mux interface moved from `core` to `optional_plugins`
* 13316b0059c9dff1634c6f062f3836f5cab530de - split varianter and parameters modules
* python3 support
* `avocado variants` deprecated arguments removed
* 73c57528b628297e6a45c4d4d614bbddb39aa8a7 - stripped and define the test state content
* 51e46987af536048b55853ffdbfbde62b62cb822 - many internal renames from implementation to concept (`mux` -> `variants`)
* 2be50b1d6fd85c61d6592ac3289b4408e8b426e9 - find free port randomly
* 5bd8263644835acca85a498af0db5de8efe0208a - ordered tests discovery
* 35f2ec8af42837da22e7f54c612c5c7803fcfecc - wheel packages support
* 29449cc9319189b211c935812b68ac71ac552ad4 - different success-rate in html plugin
* 1f296fac5af714d2008b5bb5ea934c7829317bbe - legacy interface dropped
* 3647cae545a82edeb618743b82156d9adc87ee72 - utils.kernel - support for other than 3.x versions
* d639baf30ca5db492d9b1e34a58c919d70c19bde - DummyLoader moved to optional plugins
* ae272b29c6a751cb850c727c39fa0823460e63c4, 3eab6f58f8c2f77ef36f75d2108ec41ea4be3da4 - refinements to `SimpleTest` and `ExternalRunner` to avoid using `name` for commands, because test name can be altered by plugins (adjust your custom loaders to avoid similar issues)
* d716b018a8069a2a57bbddb63d9afe01c24a21e9 -`varianter` supports `__len__`

clebergnu commented:
@ldoktor all your comments here are valid, but the approach I chose (and believe to be the right one at this stage) is to list the changes that are relevant to the next LTS.  Then, content will be massaged, squashed, reorganized, removed, etc, when the time comes to write that specific release notes.

I'm pointing that out because those reflect (are a copy) of the individual versions release notes, and I don't think we should be updating the old release notes (unless there are mistakes such as `self.parameters.get`).

Also, when it comes to docs, I believe that another PR on top of someone else's, are better way to achieve the desired end result (it's easier to write an improved version than to suggest one by smaller bits, and for the other side to realize the suggestions). So, I'm thinking about sending a v2 with the fix only, and then, we can hold on to the LTS sprint to polish the resulting doc (with as many rounds of updates as needed).  Thoughts?

ldoktor commented:
Well there are 3 things I wanted to solve in the review. First there are some minor issues (like the `parameters`) and perhaps some little adjustments and improvements (like using the `avocado.Test` to define the context).

The second are the several items I found in log but did not see in here.

The third thing was about the structure and that is out of the scope of this PR and was just to perhaps stop and think whether we want to go that way. The result for that should be trello card (or nothing) and not update of this PR.

So please do send v2 with minor tweaks, feel free to add (some) of the missing items or please copy them into trello so we can address it in following PRs and consider the 3rd suggestion. If you think better structure would be useful please create a trello card where you merge your opinions and what I suggested here (or let me know and I'll do it).
